### DISCUSSION TITLE
List of core crates

### DISCUSSION THREAD

**Discussion by alper** [rust-lang/crates.io#8317] [Feature Requests]
Hi,

Would it be possible to make a category/flag for crates that lists the "core crates" you would be likely to need in many situations? Ideally they would have some kind of stability guarantee or an acknowledgement that they're critical to the ecosystem to the point where their breakage would be highly undesirable and have lots of side effects.

I'm thinking of a list of < 50 crates containing things like serde, rand, itertools etc. The list could be gathered from statistics and/or by curation.

This was suggested by me as a process based fix for the standard library issue and people seem to dig the idea: https://lobste.rs/s/lfg3lj/zig_rust_other_languages#c_tuzfhw

Thanks,
Alper

**Comment by Turbo87** [comment]
there are plenty of unofficial lists like that (e.g. https://github.com/rust-unofficial/awesome-rust). we don't want to get into the business of picking winners by curation and you can already sort the search results by download numbers.

**Reply by alper** [reply]
Awesome lists always become a grab bag of stuff without curation (as this one also has become).

**Comment by carols10cents** [comment]
Also see [the Rust Cookbook](https://rust-lang-nursery.github.io/rust-cookbook/), which provides code and crates for common tasks.

You can also start your own list of crates you've found to be useful and publicize that! People have done that in the past. See, for example, [stdx](https://users.rust-lang.org/t/stdx-the-missing-batteries-of-rust/2015) and all the discussion around it.

But overall, there are many different contexts and tradeoffs, so it's hard to make one set of recommendations that will likely work for everyone (and keep that list up-to-date, etc).

If this is an area you're interested in working on, it'll need to go through [the Rust RFC process](https://github.com/rust-lang/rfcs), so working on that could be your next step. It's not likely that the crates.io team will have the time or people to maintain anything like this in the near future, though.

**Reply by alper** [reply]
> Also see [the Rust Cookbook](https://rust-lang-nursery.github.io/rust-cookbook/), which provides code and crates for common tasks.

That's nice but it does not seem very strong or complete?

> example, [stdx](https://users.rust-lang.org/t/stdx-the-missing-batteries-of-rust/2015) and all the discussion

That's nice but seems entirely dead.

> it'll need to go through [the Rust RFC process](https://github.com/rust-lang/rfcs),

It seems there's hardly enough interest to maintain any of these lists. I thought it could be useful to allow anybody to create collections of packages on crates.io (circumventing the curation problem) but given how thin the website is, I'm not sure that's appropriate or whether enough people would be interested in using that.

**Comment by carols10cents** [comment]
Also there's the [list of crates that are included in the playground according to this policy](https://github.com/rust-lang/rust-playground/blob/main/CRATE_POLICY.md) that might be interesting to you.
